bin/node/runtime/src/lib.rs

```diff
@@ -554,6 +554,19 @@ sp_npos_elections::generate_solution_type!(
 pub const MAX_NOMINATIONS: u32 =
 	<NposCompactSolution16 as sp_npos_elections::CompactSolution>::LIMIT as u32;
 
+/// The numbers configured here should always be more than the the maximum limits of staking pallet
+/// to ensure election snapshot will not run out of memory.
```

Comment:
I dont quite see how we can ensure this is always the case given the variability on chain. 

I get if this is a temp solution, but i think we need to be more careful than to just add this code and forget about it later.

Reply:
Actually, this comment no longer holds exactly. Given #9285, I had to `#[ignore]` these benchmarks again, so they are not always executed (which was my intention). My intention was that we always run them as we grow these parameters and it would automatically prevent us from making them too big. 

Currently I am benchmarking only `create_snapshot`. If we do the same with `on_initialize` (which is only a smidgen different), and if it works here, I think then we can be pretty sure that the same success will hold on-chain as well. 

> how we can ensure this is always the case given the variability on chain.

Which part of it doesn't add up, and what vulnerability you mean?

Reply:
The word was variability, that being that this `25_000` value can be changed on-chain, and nothing is really updating the value here in that case.

If we have some hardcoded value here, maybe we need to pass that hard-coded value through to the pallet and make sure the on-chain value is not set higher than this hard-coded value.

Reply:
since staking's 25_000 is a storage item and it can be changed on the fly, this is not possible. This will be some manual process that we run to ensure some new value that we might want to set in staking's storage is safe.

Reply:
My thought is that we can also place a limit to the value set in storage, so it does not reach past any limit we have not prepared benchmarks for